### PR TITLE
feat: create Lobby Screen

### LINKED ISSUE #9: Create lobby screen
Before the game starts, we must create a lobby of players. After creating the lobby, players should be able to join by entering the unique ID of the lobby on the join screen.

## Sketch

Image

## Definition of done

- [x] UI of screens is implemented
- [x] Navigation between create and join screen
- [x] Navigation buttons in the main activity
- [x] Create lobbies on the backend (Optional)

### PR DESCRIPTION
**Pull request by FRALEXAlexander**
# Closes
#9

### CODE REVIEW

File: app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    fun btnMainActivity(view: View) {
+        startActivity(Intent(this, MainActivity::class.java))
+    }
+
+    fun btnCreateLobby(view: View) {
+        cLobby(null)
+    }
+
+    private fun cLobby(response: Response?) {
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt:44]: We could set a default value here so that we don't have to pass null in `btnCreateLobby`

File: app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        val lobbyName = findViewById<EditText>(R.id.inputName)
+        val lobbyIsPublic = findViewById<CheckBox>(R.id.checkBoxIsPublic)
+        val lobbyMaxPlayers = findViewById<Slider>(R.id.sliderMaxPlayers)
+        val lobbyToCreate = LobbyCreate(lobbyName.text.toString(), if (lobbyIsPublic.isChecked) 1 else 0 , lobbyMaxPlayers.value.toInt())
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt:48]: Maybe add some validation (lobby name not empty)

File: app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        Log.d("Lobby", response.toString())
+
+        if (response.isSuccessful) {
+            response.body?.string()?.let {
+                Log.d("Lobby", it)
+                val intent = Intent(this, LobbyActivity::class.java)
+                intent.putExtra("lobbyCode", it)
+                startActivity(intent)
+            }
+        } else {
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt:73]: If we get an error like the server not being online, won't we have an infinite loop of calling `cLobby`?

File: app/src/main/java/at/aau/serg/models/JoinLobbyLobby.kt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package at.aau.serg.models
+
+data class JoinLobbyLobby(val name: String, val currentPlayers: Int, val maxPlayers: Int,val lobbyID: String)
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/models/JoinLobbyLobby.kt:3]: Please change the name so we don't repeat Lobby

File: app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import at.aau.serg.logic.StoreToken
+import at.aau.serg.models.JoinLobbyLobby
+import at.aau.serg.models.LobbyJoin
+import at.aau.serg.network.CallbackCreator
+import at.aau.serg.network.HttpClient
+import com.google.gson.Gson
+import okhttp3.Response
+
+
+class JoinLobbyLobbiesAdapter(context: ContextWrapper, listdata: MutableList<JoinLobbyLobby>) :
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt:26]: Can we also change the name to something like `LobbyAdapter`

File: app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    private fun joinLobby(lobbyToJoin: LobbyJoin) {
+        HttpClient.post(
+            "/lobbys/join",
+            Gson().toJson(lobbyToJoin),
+            StoreToken(this).getAccessToken(),
+            CallbackCreator().createCallback(::onFailureLobby, ::onSuccessJoinLobby)
+        )
+    }
+
+    private fun onFailureLobby() {
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt:127]: We should probably add some more specific error handling, but we could also do that in a separate PR where we change the ` createCallback`  to include the error.

File: app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+                Log.d("Lobby", it)
+                val intent = Intent(this, LobbyActivity::class.java)
+                intent.putExtra("lobbyCode", it)
+                startActivity(intent)
+            }
+        } else {
+            HttpClient.get(
+                "/lobbys/leave",
+                StoreToken(this).getAccessToken(),
+                CallbackCreator().createCallback(::onFailureLobby, ::jLobbyWithCode)
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt:147]: This could also produce an infinite loop if there is an error unrelated to the player being in a lobby

File: app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                Log.d("Lobby", it)
+                val intent = Intent(context, LobbyActivity::class.java)
+                intent.putExtra("lobbyCode", it)
+                context.startActivity(intent)
+            }
+        } else {
+            HttpClient.get(
+                "/lobbys/leave",
+                StoreToken(context).getAccessToken(),
+                CallbackCreator().createCallback(::onFailureLobby, ::joinLobby)
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt:83]: Could also produce a loop

File: app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        return ViewHolder(listItem)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val lobby: JoinLobbyLobby = listdata[position]
+        holder.txtLobbyName.text = lobby.name
+        holder.txtPlayerCount.text = context.getString(R.string.joinlobbyLobbyPlayerCountPlaceholder, lobby.currentPlayers,lobby.maxPlayers);
+        holder.btnJoin.setOnClickListener {
+            lobbyID = lobby.lobbyID
+            joinLobby(null)
```
Comment (TannerGabriel) [inline app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt:52]: We don't get an error message when we try to join a lobby that is already full

**Review by TannerGabriel — CHANGES_REQUESTED** [review]
Looks good. I left some minor comments

File: app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt (hunk 6 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt:127]: Yes i would move that to the other pr

File: app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt (hunk 5 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt:26]: The name is debateble this version is more despriptive yours is shorter both would work for me

File: app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt (hunk 1 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt:44]: I would do that in the pr related to the callbackcreater and the error handling of it

File: app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt (hunk 3 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/activities/CreateLobbyActivity.kt:73]: Will be fixed in #21

File: app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt (hunk 7 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt:147]: Will be fixed in #21

File: app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt (hunk 8 above)
Comment (FRALEXAlexander) [reply app/src/main/java/at/aau/serg/adapters/JoinLobbyLobbiesAdapter.kt:83]: Will be fixed in #21

**Review by TannerGabriel — APPROVED** [review]
